### ISSUE TITLE
ocaml build missing calendar dependency

### ISSUE THREAD

**Issue by dckc** [dmbaturin/soupault#2]
The README says to run `opam pin add soupault .` but the result I get is:

```
[ERROR] Opam has not been initialised, please run `opam init'
```
So I run `opam init` and accept the defaults and try again. This time, it builds a bunch of stuff

```
...
 installed ppx_monadic.2.3.0
[ERROR] The compilation of soupault failed at
        "/home/connolly/.opam/opam-init/hooks/sandbox.sh build dune build -p
        soupault".

#=== ERROR while compiling soupault.0.1 =======================================#
# context     2.0.4 | linux/x86_64 | ocaml-system.4.05.0 | pinned(git+file:///home/connolly/projects/soupault#master#8f2cb1ae)
# path        ~/.opam/default/.opam-switch/build/soupault.0.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p soupault
# exit-code   1
# env-file    ~/.opam/log/soupault-24248-96e30a.env
# output-file ~/.opam/log/soupault-24248-96e30a.out
### output ###
# File "src/dune", line 4, characters 76-84:
# 4 |  (libraries lambdasoup toml containers fileutils logs logs.fmt stringext re calendar yojson)
#                                                                                 ^^^^^^^^
# Error: Library "calendar" not found.
# Hint: try: dune external-lib-deps --missing -p soupault @@default


<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build soupault 0.1
└─ 

```

So I try `dune external-lib-deps ...` and it installs `calendar.2.04` and I try again... oh. I guess it worked this time. I'm not sure where the binary went, but I guess it's working as expected:

```
$ soupault
soupault: [WARNING] Configuration file soupault.conf not found, using default settings
Error: templates/main.html: No such file or directory
```

The [install docs](https://baturin.org/projects/soupault/#installation) say ...

> If you are familiar with the OCaml programming language, you may want to install from source.

Well, I suppose there are some OCaml build conventions that I'm not familiar with. Care to point me to some clues?

https://opam.ocaml.org/doc/Manual.html seems to be about opam commands that grab things from the cloud, not about building from source.

**Comment by dmbaturin** [comment]
No, this is my fault. I forgot to add the calendar library to the opam package dependencies.

Myself I usually just run `dune build` rather than reinstall the package, and in my setup opam deps are not pulled from the build script (`src/dune`), so this slipped through the cracks.

I've pushed the fix, please try again.

**Comment by dckc** [comment]
yes, it's fixed in c1417cb6c3e1 . Thanks!

**Comment by dmbaturin** [comment]
...the binary goes to ~/.opam/$switch_name/bin
The cool thing about opam is that you can keep multiple compiler versions/variants and switch between them with "opam switch $switch_name".

The error about template, I should probably check if site dir exists and give a more indicative error and a suggestion to run `soupault --init`, plus output a link to the docs.
